File: bench/fib_loop_big.java
// BOT-TRANSLATED

import java.math.BigInteger;

final class fib_loop_big {
  static final BigInteger WANT =
    new BigInteger("205697230343233228174223751303346572685");
  static volatile int depth = 184;
  static volatile int runs = 1 << 21;

  static BigInteger fib(int depth) {
    BigInteger prev = BigInteger.ZERO;
    BigInteger next = BigInteger.ONE;

    while (depth-- > 0) {
      final BigInteger tmp = prev.add(next);
      prev = next;
      next = tmp;
    }

    return next;
  }

  static void run(int runs) {
    BigInteger out = BigInteger.ZERO;

    while (runs-- > 0) out = fib(depth);

    if (!out.equals(WANT)) {
      throw new AssertionError("big iterative Fibonacci output mismatch");
    }
  }

  public static void main(String[] args) {
    run(runs);
  }
}

File: bench/fib_loop_big.py
# BOT-TRANSLATED

def fib(depth):
  prev = 0
  next = 1

  while depth > 0:
    prev, next = next, (prev + next)
    depth -= 1

  return next

def run(depth, runs):
  out = 0
  while runs > 0:
    out = fib(depth)
    runs -= 1
  if out != 205697230343233228174223751303346572685:
    raise RuntimeError("big iterative Fibonacci output mismatch")

run(184, 1 << 21)
